protos/flyteidl/admin/description_entity.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package flyteidl.admin;
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin";
+
+import "flyteidl/core/identifier.proto";
+import "flyteidl/admin/common.proto";
+
+// DescriptionEntity contains detailed description for the task/workflow/launch plan.
+// Documentation could provide insight into the algorithms, business use case, etc.
+message DescriptionEntity {
+  // One-liner overview of the entity.
+  string short_description = 1;
+  // Full user description with formatting preserved.
+  LongDescription long_description = 2;
+  // Optional link to source code used to define this entity.
+  SourceCode source_code = 3;
+  // User-specified tags. These are arbitrary and can be used for searching
+  // filtering and discovering tasks.
+  repeated string tags = 4;
+}
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
+  oneof content {
+    // long description - no more than 4KB
+    string value = 1;
+    // if the description sizes exceed some threshold we can offload the entire
+    // description proto altogether to an external data store, like S3 rather than store inline in the db
+    string uri = 2;
+  }
+
+  enum DescriptionFormat {
+    UNKNOWN = 0;
+    MARKDOWN = 1;
+    HTML = 2;
+    // python default documentation - comments is rst
```

Comment:
Interesting. Will this be a problem for rendering in the Hub? I thought we are expecting markdown.. @wild-endeavor @jsonporter

Reply:
i feel like maybe we can remove this comment actually.  but i don't think it's fair to assume one or the other.  we'll probably have to parse and take a guess.  i'm expecting people to write in both styles, but internally to flyte, all our python comments are written as sphinx rst.

Reply:
Mind chatting with Jason about this? Not sure if we will need to translate from one to another or if we can have different renderers... having the enum here is actually useful to determine that behavior